### PR TITLE
feat: Controller Leader Election

### PR DESCRIPTION
**Pull request by davidcollom**
This implements leader Election for the argo-events controller as mentioned in #1565 

Closes: #1565

### CODE REVIEW

File: manifests/cluster-install/rbac/argo-events-cluster-role.yaml [hunk 1]
```diff
@@ -3,6 +3,20 @@ kind: ClusterRole
 metadata:
   name: argo-events-role
 rules:
+  - apiGroups:
```
Comment (whynowy) [inline manifests/cluster-install/rbac/argo-events-cluster-role.yaml:6]: This rules need to be refined. For example, `configmaps` has been defined in line 46.

Comment (davidcollom) [reply manifests/cluster-install/rbac/argo-events-cluster-role.yaml:6]: Did not see `configmap` 🤦  - Will update.

**Comment by whynowy** [comment]
I checkout your branch, and tried on my local, seeing following in the controller log - shall we add the `events` to the RBAC?

```
E0422 18:16:44.746332       1 event.go:267] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"argo-events-controller.16e84ab503810658", GenerateName:"", Namespace:"argo-events", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Lease", Namespace:"argo-events", Name:"argo-events-controller", UID:"fa83aada-083b-488d-aa95-b78ae2ab55ea", APIVersion:"coordination.k8s.io/v1", ResourceVersion:"7702868", FieldPath:""}, Reason:"LeaderElection", Message:"controller-manager-6fdc445ff8-plmn6_d6524b84-cdec-4877-a9fc-bd861a358363 became leader", Source:v1.EventSource{Component:"controller-manager-6fdc445ff8-plmn6_d6524b84-cdec-4877-a9fc-bd861a358363", Host:""}, FirstTimestamp:time.Date(2022, time.April, 22, 18, 16, 44, 739610200, time.Local), LastTimestamp:time.Date(2022, time.April, 22, 18, 16, 44, 739610200, time.Local), Count:1, Type:"Normal", EventTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'events is forbidden: User "system:serviceaccount:argo-events:argo-events-sa" cannot create resource "events" in API group "" in the namespace "argo-events"' (will not retry!)
```

**Review by whynowy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**